elementary/monitor/api/tests/tests.py

```diff
@@ -41,35 +41,32 @@ def __init__(self, dbt_runner: DbtRunner):
         super().__init__(dbt_runner)
         self.invocations_api = InvocationsAPI(dbt_runner)
 
-    @staticmethod
-    def get_test_sub_type_unique_id(
-        model_unique_id: str,
-        test_unique_id: str,
-        column_name: Optional[str] = None,
-        test_sub_type: Optional[str] = None,
-        **kwargs,
-    ) -> str:
-        return f"{model_unique_id}.{test_unique_id}.{column_name if column_name else 'None'}.{test_sub_type if test_sub_type else 'None'}"
-
-    def get_tests_metadata(
+    def _get_all_test_results(
         self,
         days_back: Optional[int] = 7,
-        invocation_id: str = None,
+        invocations_per_test: int = 720,
+        metrics_sample_limit: int = 5,
+        disable_passed_test_metrics: bool = False,
         should_cache: bool = True,
-    ) -> List[TestMetadataSchema]:
+    ) -> List[RawTestResultSchema]:
         run_operation_response = self.dbt_runner.run_operation(
             macro_name="get_test_results",
-            macro_args=dict(days_back=days_back, invocation_id=invocation_id),
+            macro_args=dict(
+                days_back=days_back,
+                invocations_per_test=invocations_per_test,
+                metrics_sample_limit=metrics_sample_limit,
+                disable_passed_test_metrics=disable_passed_test_metrics,
+            ),
         )
-        tests_metadata = (
+        test_results = (
             json.loads(run_operation_response[0]) if run_operation_response else []
         )
-        tests_metadata = [
-            TestMetadataSchema(**test_metadata) for test_metadata in tests_metadata
+        test_results = [
+            RawTestResultSchema(**test_result) for test_result in test_results
         ]
         if should_cache:
```

Comment:
I personally think that we should delete this cache mechanism and create a coherent data flow in the program instead.

Reply:
Deleted!